lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -87,126 +88,100 @@ bool connectShaderToMaterial(
 #endif
 } // namespace
 
-UsdPrim BindMaterialUndoableCommand::CompatiblePrim(const Ufe::SceneItem::Ptr& item)
+bool BindMaterialUndoableCommand::CompatiblePrim(const Ufe::SceneItem::Ptr& item)
 {
     auto usdItem = std::dynamic_pointer_cast<const MAYAUSD_NS::ufe::UsdSceneItem>(item);
     if (!usdItem) {
-        return {};
+        return false;
     }
     UsdPrim usdPrim = usdItem->prim();
     if (UsdShadeNodeGraph(usdPrim) || UsdShadeShader(usdPrim)) {
         // The binding schema can be applied anywhere, but it makes no sense on a
         // material or a shader.
-        return {};
+        return false;
     }
     if (UsdGeomScope(usdPrim) && usdPrim.GetName() == kDefaultMaterialScopeName) {
-        return {};
+        return false;
+    }
+    if (auto subset = UsdGeomSubset(usdPrim)) {
+        TfToken elementType;
+        subset.GetElementTypeAttr().Get(&elementType);
+        if (elementType != UsdGeomTokens->face) {
+            return false;
+        }
     }
     if (PXR_NS::UsdShadeMaterialBindingAPI::CanApply(usdPrim)) {
-        return usdPrim;
+        return true;
     }
-    return {};
+    return false;
 }
 
 BindMaterialUndoableCommand::BindMaterialUndoableCommand(
-    const UsdPrim& prim,
+    Ufe::Path      primPath,
     const SdfPath& materialPath)
-    : _stage(prim.GetStage())
-    , _primPath(prim.GetPath())
+    : _primPath(std::move(primPath))
     , _materialPath(materialPath)
 {
 }
 
 BindMaterialUndoableCommand::~BindMaterialUndoableCommand() { }
 
-void BindMaterialUndoableCommand::undo()
+void BindMaterialUndoableCommand::undo() { _undoableItem.undo(); }
+
+void BindMaterialUndoableCommand::redo() { _undoableItem.redo(); }
+
+void BindMaterialUndoableCommand::execute()
 {
-    if (!_stage || _primPath.IsEmpty() || _materialPath.IsEmpty()) {
+    if (_primPath.empty() || _materialPath.IsEmpty()) {
         return;
     }
```

Comment:
IDk if it warrants a warning or some other user feedback when the arguments are not valid?

Reply:
If the error handling scheme is still `CTOR throws if command cannot execute` then I agree this test should be moved to the constructor and throw accordingly. The execute() function should assume these are valid. Will fix that this afternoon.